Library/Homebrew/dev-cmd/extract.rb

```diff
@@ -142,6 +143,15 @@ def extract
         test_formula = formula_at_revision(repo, name, file, rev)
         break if test_formula.nil? || test_formula.version == version
 
+        test_version = [version, test_formula.version]
+        if test_version.reduce { |r, v| r && Gem::Version.correct?(v) }
```

Comment:
Why use `.reduce` here and discard the result? Can you explain the logic you're trying to do here?

Reply:
Check the format is correct before parsed by `Gem::Version`

Reply:
In that case I'd refactor this whole block to just use `version` and `test_formula.version` directly rather than using this array and using `test_version.first` and `test_version.last`. e.g. 

```suggestion
        if version.present? && Gem::Version.correct?(version) && 
           test_formula.version.present? && Gem::Version.correct?(test_formula.version.present)
```

Note I have not tested any of these suggested changes

Reply:
I'll do the rest ;)